include/nova_renderer/window.hpp

```diff
@@ -38,14 +38,30 @@ namespace nova::renderer {
          * This callback will key called for every key press event that this window receives
          *
          * \param key_callback Callback for when a key is received. Intentionally a std::function so I can easily add
-         * it to a vector
+         * it to a vector. First parameter to this function is the key code, second is whether the key was pressed
+         * this frame, third is if control is down, fourth is if shift is down
          */
-        void register_key_callback(std::function<void(uint32_t)>&& key_callback);
+        void register_key_callback(std::function<void(uint32_t, bool, bool, bool)>&& key_callback);
```

Comment:
Why are you taking these by rvalue reference? This probably should just be by value, as you want both rvalues and lvalues to be able to call it, and you're going to copy it anyway.

Reply:
idk it seemed kinda fun

Reply:
One note, when you do write it with a copy, the insertion into the vector needs to be a move.